[BLOG] July community update (#219)
pulsar-edit/pulsar-edit.github.io: docs/blog/20230701-Daeraxa-JulyUpdate.md

---
title: Community Update
author: Daeraxa
date: 2023-07-01
category:
  - news
  - log
tag:
  - update
---

What has it got in its pocketses? It's the July community update!

<!-- more -->

# Welcome to the July Community Update

This month's update is a maybe a little different in tone to some of the others in that we have had to deal with some problems recently outside of our control which means our first few items are a little more "serious" and a bit less optimistic that our usual updates. That being said these have been front loaded into this update with all the good news coming in the second half, so don't despair!

> This month we have to cover some of the issues such as our CI not building binaries, discussions about the future of our subreddit and some false antivirus positives on our packages but we then come back strong with a new method of downloading Pulsar and a bunch of improvements to the application!

Author: Again I'll pick from this as I kind of abhor the "Oxford comma" unless being used to remove ambiguity. I'm also really not keen on starting sentences with "but", I know it isn't a hard rule but in this case I find it kind of jarring.
I also think it should be restructured to allow "cleanse the palate" rather than "cleans".

> So with all that covered lets get on with it!

Author: I think I'd prefer to leave it (with the correction of course)

## ARM builds problem

<img src="./assets/Linux-ARM.jpg" height=200>

> Some of you may have noticed that we had a bit of a problem recently in producing binaries for ARM on Linux which, whilst for the most part shouldn't have affected anyone except ARM Linux users, did have a knock on effect with some of the community maintained packages on certain package managers. Mostly this would have just been a delay in updating the package to 1.106.0.

Author: Again I think I want to leave out "package managers" intentionally to make sure we are covering everything. I'm also not sure about the shortening the sentences but that goes back to the prosaic vs conversational discussion.

This wasn't anything to do with our code or builds but was instead a failure of the CI machine used to produce those builds failing to properly connect to various services.

> This has now been resolved so if you were waiting for an ARM binary then these are fully available on our [downloads](https://pulsar-edit.dev/download.html) page as normal and for those relying on a community package or distribution then any blocker will now have been cleared.

Author: I intentionally didn't add package manager as I wanted to make sure that non-package managed redistributed binaries are also covered by this statement

We recently decided to [re-open the subreddit](https://www.reddit.com/r/pulsaredit/comments/14ktr3q/rpulsaredit_is_reopening_read_on_for_our/) for a number of reasons, some because we don't want to abandon a portion of our community entirely and partly because of threats from Reddit themselves.

The reason I bring this up here is because it is already possible that some people who were using Reddit for Pulsar news have now left it and won't have seen the above post. This post also contains a poll to get feedback from the affected community to see what our next steps would be. Should we close the subreddit and abandon it? Should we open a Lemmy community instead? Should we do both?

So if you are interested in either using Reddit or Lemmy then please have a look at the above post and vote in the poll.

## Antivirus `es5-ext` issue

<img src="./assets/malware.png" height=200>

Some users have been reporting that various antivirus applications and virus scanning tools have been identifying Pulsar as a virus due to the `es5-ext` package.

This is is nothing to worry about, Pulsar is not a virus nor do we use dependencies that are.

> After some investigation we discovered that somewhere in our dependency tree (i.e. a package dependency of a package we use) exists this `es5-ext` package. At some point this package was updated to display a "message of peace" to Russians (in reference to the ongoing conflict). This message displays to anyone within the Russian timezones and, whilst the "payload" of a simple message is harmless, the act of targeting based on time-zone has caused this package to be flagged by various scanners and the package author does not seem to want to remove it.

Author: I feel this leaves the "This message displays" sentence way too short so I'll pick from this one.

> So we have been working on a way to get around this and [@confused-techie] has managed to [pin this to an earlier version](https://github.com/pulsar-edit/pulsar/pull/608) using our own fork of this package.

Reviewer: ```suggestion
So we have been working on a way to get around this and [@confused-techie] with the help of [@savetheclocktower] has managed to [pin this to an earlier version](https://github.com/pulsar-edit/pulsar/pull/608) using our own fork of this package.
```

Also not sure about wording, but while originally I tried to pin to an older unaffected version, or fork is the latest of this package, but removes the offending code. So maybe "has managed to fork the package and remove the offending code that was being flagged as malicious"

Hopefully this should provide some peace of mind to anyone who has encountered this and with our update this should no longer cause people to worry.

## Pulsar available on deb-get

<img src="./assets/deb-get-logo.png" height=200>[^2]

> So I think it is of no surprise that we want to get Pulsar on various package managers but with things developing so quickly and with us wanting to make sure we can support some of the more basic features with our current builds we have been taking this somewhat slowly.

Author: Addressed this but with a few changes.

> This is really just a small announcement to say that Pulsar is officially supported on [deb-get](https://github.com/wimpysworld/deb-get) which is a package manager for Debian that is perfect for us at this point as all it really does is grab a pre-built `.deb` binary from our GitHub releases and install it as any other `.deb`. Our main reason to support this is because it doesn't need to do any setting of your environment or need any maintenance at all - it is just a small and simple script that just goes and grabs the latest version - you can see it on the [deb-get repo](https://github.com/wimpysworld/deb-get/blob/main/01-main/packages/pulsar)

Author: I've made some updates here but mostly picked from this.

However this does mean you can now easily manage it. For example, once you have installed `deb-get` all you have to do to install Pulsar is:

- `deb-get install pulsar`

To update Pulsar (and other `deb-get` packages, all you need to do is:

- `deb-get update`
- `deb-get upgrade`

So if you are on a Debian based distro and want a quick and easy way to install and update Pulsar, give it a try!

## Tree-sitter migration of TOML grammar

<img src="./assets/toml.png" height=200>[^3]

In our [last regular release](https://pulsar-edit.dev/blog/20230616-Daeraxa-v1.106.0.html) we made a big deal about our modern Tree-sitter migration and new grammars that come with it.

> This work is still very much ongoing and we have new grammars being migrated all the time - our latest is the [TOML Tree-sitter grammar](https://github.com/pulsar-edit/pulsar/pull/617) that [@savetheclocktower] has added which come with all the benefits of a Tree-sitter grammar such as more consistent highlighting and scope for new features in the future.

Reviewer 2: ```suggestion
This work is still very much ongoing and we have new grammars being migrated all the time. Our latest is the [TOML Tree-sitter grammar](https://github.com/pulsar-edit/pulsar/pull/617) that [@savetheclocktower] has added. It's got all the benefits of a Tree-sitter grammar: more consistent highlighting, better performance, and support for features that can't be delivered with a TextMate-style grammar.
```

As ever we are after feedback, so if you use this new TOML Tree-sitter grammar (or any other for that matter) and have any issues or see any inconsistencies compared to the existing grammar then please let us know on [Discord](https://discord.gg/7aEbB9dGRT) or [file an issue](https://github.com/pulsar-edit/pulsar/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc).

## `less-cache` package update

<img src="./assets/less.png" height=200>[^4]

This is still in the works but [@confused-techie] has been making some updates to be added in the near future that bumps the version of `less-cache` that we are using.

`less-cache` is a module that handles turning all Pulsar's (and our community's) packages written in [less](https://lesscss.org/) into valid CSS, as well as ensuring each one is able to import the values from Pulsar that it cares about. For the past three years `less-cache` has been using `less@3.12.2` which, while fine, has been missing out on some of the big new changes in less after version 4 was released.

> So [this PR](https://github.com/pulsar-edit/pulsar/pull/611) bumps it to `less-cache@2.0.0` in order to bump the version of `less` to 4.1.3 which brings along these exciting new features but also introduces breaking changes to existing `less` stylesheets. The biggest of which being:

Reviewer 2: ```suggestion
So [this PR](https://github.com/pulsar-edit/pulsar/pull/611) bumps it to `less-cache@2.0.0` in order to bump the version of `less` to 4.1.3. This change gives us some exciting new features, but it also introduces breaking changes to existing `less` stylesheets:
```

> - Parantheses required for mixin calls
> - Parens-division now the default math setting

Reviewer 2: ```suggestion
- Parentheses required for mixin calls (`.some-mixin()` instead of `.some-mixin`)
- Parentheses now required around math expressions that use division
```

Reviewer: Yeah the typo here is my bad, so whoops!

But I do want to add the other than the typo, this message is identical to what was included on the [`v4.0.0` release notes](https://github.com/less/less.js/releases/tag/v4.0.0), so things like `Parens-division` don't have to be changed, but if we feel they are more understandable changed, then that's totally fair

Author: Yeah if it is on the release notes it should be retained here (also my british brain just wants to get rid of "parentheses" entirely but I know this should be left :P )

> The most noticeable thing, at least within Pulsar, is that all division is required, and all calculations are recommended, to be enclosed within parentheses. Essentially meaning `less` does math less eagerly, as opposed to 3.x which is described as doing math eagerly.

Reviewer 2: ```suggestion
The latter item has more impact: because `/` is used in newer CSS features like grids, parentheses are needed when doing division to remove ambiguity about the function of `/` on a particular line. Lots of usages in Pulsar core needed to be updated as a result of this, so it's quite possible that third-party packages are affected as well.
```

> So existing stylesheets in syntax themes, UI themes and community packages need to be updated to ensure that `less` properly does the math rather than handing off invalid CSS, with the math still included. At times an error may be shown informing the user the stylesheet failed to be compiled, whereas in some cases, it won't be reported and instead can result in a broken UI. Pulsar itself has already been updated, as well as all core packages, but now we have to keep a lookout for non-archived community packages that may need help with this change.

Reviewer 2: ```suggestion
We're brainstorming ways that we can detect these usages and minimize the impact on these packages. But if you've got a community package that you're maintaining, and it uses`less` stylesheets, please take a moment to see if it's affected. At times an error may be shown informing the user that the stylesheet failed to be compiled; in other cases, it won't be reported and instead can result in a broken UI.
```

Reviewer: Thanks for adding this! I know this is a newer development, that was thought of after the original thing was written, so appreciate it!

> Despite these issues it does mean that we can now take advantage of the new `less` features, and be able to keep our implementation up to date, as well as being able to use any modern `less` examples and docs to create styles

Reviewer 2: ```suggestion
The upside of this upgrade is that we can take advantage of new `less` features and keep our implementation up to date.
```

---

> And just like that we are done with yet another month's updates. As ever if you want to get more involved the community then feel free to join in on our various [social channels](https://pulsar-edit.dev/community.html). Hope to see you again this time next month!

Reviewer 2: ```suggestion
And just like that, we are done with yet another month's updates. As ever, if you want to get more involved in the community, please join in on our various [social channels](https://pulsar-edit.dev/community.html). Hope to see you again this time next month!
```

Author: Thanks, definitely prefer this language to actively invite people in.

So if you are interested in using either Reddit or Lemmy, please have a look at the above post and vote in the poll.

## Antivirus `es5-ext` issue

<img src="./assets/malware.png" height=200>

Some users have been reporting that various antivirus applications and virus scanning tools have been identifying Pulsar as a virus due to the `es5-ext` package.

This is is nothing to worry about — Pulsar is not a virus, nor do we use dependencies that are.

After some investigation, we discovered that somewhere in our dependency tree (i.e., a package dependency of a package we use) exists this `es5-ext` package. Early last year, this package was updated to display a "message of peace" to Russians (in reference to the ongoing conflict). This message displays to anyone within a Russian time zone, and, whilst the "payload" of a simple message is harmless, the act of targeting based on time zone has caused this package to be flagged by various scanners. Unfortunately the package author does not seem to want to remove it.

> So we have been working on a way to get around this. [@confused-techie] has managed to [pin this to an earlier version](https://github.com/pulsar-edit/pulsar/pull/608) using our own fork of this package.

Reviewer: ```suggestion
So we have been working on a way to get around this. [@confused-techie], with [@savetheclocktower]'s help, has managed to [pin this to an earlier version](https://github.com/pulsar-edit/pulsar/pull/608) using our own fork of this package.
```

Reviewer: @Daeraxa This is the one that got lost

In our experiments, binaries built in this manner are no longer being flagged by any notable antivirus program. Hopefully this gives our users peace of mind.

## Pulsar available on deb-get

<img src="./assets/deb-get-logo.png" height=200>[^2]

> We want to put Pulsar on all major package managers, but it can be trickier than it looks. Various package managers have special requirements and restrictions, so it behoves us to move slowly and make sure we understand those constraints.

Reviewer 2: ```suggestion
We want to put Pulsar on all major package managers, but it can be trickier than it looks. Various package managers have special requirements and restrictions, so it behooves us to move slowly and make sure we understand those constraints.
```

Author: This isn't a typo?

Reviewer: Apparently:

>“Behoves” and “behooves” are both correct spellings of the word “behoove”. “Behove” is the British English spelling.

Author: Well that would explain it lol

Reviewer: Yeah, so I'd say since you normally try to go for an EN-GB style here, lets leave it at `behooves`

Reviewer 2: That's a new regional variant for me. I now have no opinion on the outcome here. :)
